src/cli/rustup_mode.rs

```diff
@@ -84,8 +85,33 @@ where
     callee(cfg, matches)
 }
 
+async fn deprecated_async<'a, FF, F, B, R>(
+    instead: &str,
+    cfg: &'a mut Cfg,
+    matches: B,
+    callee: FF,
+) -> R
+where
+    FF: FnOnce(&'a mut Cfg, B) -> F,
+    F: Future<Output = R> + 'a,
+{
+    (cfg.notify_handler)(Notification::PlainVerboseMessage(
+        "Use of (currently) unmaintained command line interface.",
+    ));
+    (cfg.notify_handler)(Notification::PlainVerboseMessage(
+        "The exact API of this command may change without warning",
+    ));
+    (cfg.notify_handler)(Notification::PlainVerboseMessage(
+        "Eventually this command will be a true alias.  Until then:",
+    ));
+    (cfg.notify_handler)(Notification::PlainVerboseMessage(&format!(
+        "  Please use `rustup {instead}` instead"
+    )));
```

Comment:
Given that this is exactly the same as in sync `deprecated()`, let's deduplicate? Alternatively, is there a good reason not to use the async variant for the `uninstall` command so we don't need both?